content/en/blog/2023/end-user-discussions-03.md

```diff
@@ -0,0 +1,273 @@
+- [Tail sampling processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/tailsamplingprocessor/README.md)
+
+### ADOPTION, MIGRATION, AND IMPLEMENTATION
+
+#### 1 - Common migration challenges
+
+**Q:** What are common challenges faced by developers when migrating to
+OpenTelemetry?
+
+**More context:** We have hundreds of microservices that need to be migrated,
+including big monolith systems with a lot of custom tracing locked into specific
+vendors and their libraries. Setting up agents to facilitate this migration is
+like having two different sets of observability systems running at the same
+time.
+
+**A:** One user shared their journey: They started by using a backend that
+supports OpenTelemetry. The two challenges they faced were: a cultural change in
+the engineer’s mindset, and raising awareness of OpenTelemetry, which are bigger
+than the technical challenges. The key is to not propose one big change; the
+journey of moving from a vendor-based solution to OpenTelemetry should be a
+step-by-step process, rather than going into a full transformation.
+
+Additional suggestions:
+
+- Start with dev or testing environments first to build trust in the software
+- Choose a stack where OTel is more robust, such as Java and Node.js
+- For countering developer resistance, using auto-instrumentation modules to
+- start with is a good step
+
+#### 2 - Starting and scaling
+
+**Q:** What is a good place to start from with OpenTelemetry? For example, from
+infra to data collection, or starting in the application? And how do you scale
+it up?
+
+**More context**: Our use case is end-to-end visibility; currently, we are using
+a vendor for monitoring logs, metrics, and traces. We are also using things like
+RUM (real user monitoring). Can we do the same with OpenTelemetry, and at scale?
+
+**A:** It depends on if you are starting to use OTel in a new project, or trying
+to re-orchestrate an existing or old project. It’s best to start with a
+transition plan, make sure the performance impact is not bad, and scale up what
+you need. One suggestion is to start experimenting with Java OTel
+instrumentation, as the overall performance impact is negligible.
+
+Another suggestion is to try infrastructure monitoring with OpenTelemetry using
+the
+[host metrics receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/hostmetricsreceiver/README.md)
+in the Collector, as it covers a lot of metrics, and has no dependencies. One
+user noticed a 20% reduction in CPU usage when they moved from a vendor-specific
+agent to the host metrics receiver for infrastructure monitoring.
+
+#### 3 - Auto-instrumentation
+
+**Q:** Is there a way to automatically create spans without code changes?
+
+**A:** It depends on the use cases:
+
+- Auto instrumentation options are maturing in OTel; for example, the Java JAR
```

Comment:
Is it worth mentioning auto-instrumentation also available for Python, NodeJS, and .NET?